src/App.vue

```diff
@@ -30,60 +30,79 @@ export default {
   data() {
     return {
       onHome: true,
-      potd: {},
-      podb: {},
-      pond: {},
-      todaysDate: this.getTodaysDate(),
+      loading: '',
       error: '',
-      loading: ''
+      todaysDate: this.getDate(),
+      todayPOTD: {},
+      previousPOTD: {},
+      nextPOTD: {},
     }
   },
   methods: {
+    getPhoto(date, option) {
+      apiCalls.getSpecificDatesPhoto(date)
+        .then(results => {
+          this.handleError(results, option)
+        });
+    },
+
+    handleError(result, option) {
+      if (typeof result === 'string') {
+        this.error = result;
+
+      } else {
+        this[option] = result;
+      }
+    },
+
+    checkForDateParams(date, relation) {
+      if (date && relation) {
+        return moment(date);
+
+      } else {
+        return moment();
+      }
+    },
+
+    getDate(date, relation) {
+      let today = this.checkForDateParams(date, relation);
+    
+      if (relation === 'next') {
+        today = today.add(1, 'days');
+
+      } else if (relation === 'previous') {
+        today = today.subtract(1, 'days');
+
+      }
+
+      return today.format('YYYY-MM-DD');
+    },
+
     showPicture(date) {
-      this.loading = 'loading....';
-      if(date === undefined) {
+      if (date === undefined) {
         date = this.todaysDate;
       }
-      apiCalls.getSpecificDatesPhoto(date)
-        .then(results => {
-          if (typeof results === 'string') {
-            this.error = results
-          } else {
-            this.potd = results}
-          })
-
-      const dayBefore = this.getPreviousDate(date)
-      apiCalls.getSpecificDatesPhoto(dayBefore)
-        .then(results => {
-          if (typeof results === 'string') {
-            this.error = results
-          } else { this.podb = results}})
+
+      this.loading = 'loading....';
+
+      const dayBefore = this.getDate(date, 'previous');
+      const nextDay = this.getDate(date, 'next');
```

Comment:
Should this be called in the conditional below same as the `this.getPhoto(nextDay, 'nextPOTD');` ? That way its not being run when its not necessary

Reply:
It might need its own condition. You're trying to prevent it showing a `previous day` preview if the user is looking at the first ever POTD, right? We would just have to check that the date is not the last date the API includes I guess.

Reply:
Nevermind I see what you're saying. I'll scope it into the `if` block